terraform/jenkins/binary_cache_signing.tf

```diff
@@ -0,0 +1,47 @@
+# nix-store --generate-binary-cache-key foo secret-key public-key
+# terraform import secret_resource.binary_cache_signing_key "$(< ./secret-key)"
+resource "secret_resource" "binary_cache_signing_key" {
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+# Create an Azure key vault.
+resource "azurerm_key_vault" "binary_cache_signing_key" {
+  # this must be globally unique
+  name                = "ghaf-binarycache-signing"
+  location            = azurerm_resource_group.default.location
+  resource_group_name = azurerm_resource_group.default.name
+  sku_name            = "standard"
+  #  The Azure Active Directory tenant ID that should be used for authenticating
+  # requests to the key vault.
+  tenant_id = data.azurerm_client_config.current.tenant_id
+}
+
+# Upload the binary cache signing key as a vault secret
+resource "azurerm_key_vault_secret" "binary_cache_signing_key" {
+  name         = "binary-cache-signing-key"
+  value        = secret_resource.binary_cache_signing_key.value
+  key_vault_id = azurerm_key_vault.binary_cache_signing_key.id
+
+  # Each of the secrets needs an explicit dependency on the access policy.
+  # Otherwise, Terraform may attempt to create the secret before creating the
+  # access policy.
+  # https://stackoverflow.com/a/74747333
+  depends_on = [
+    azurerm_key_vault_access_policy.binary_cache_signing_key_terraform
+  ]
+}
+
+resource "azurerm_key_vault_access_policy" "binary_cache_signing_key_terraform" {
+  key_vault_id = azurerm_key_vault.binary_cache_signing_key.id
+  tenant_id    = data.azurerm_client_config.current.tenant_id
+  # "TerraformAdminsGHAFInfra" group
+  object_id = "f80c2488-2301-4de8-89d6-4954b77f453e"
```

Comment:
can we avoid hardcoded object-ids?

Reply:
This is a "Group" in Azure (probably an Entra Group), and I'm not sure which terraform resource can be used for that. It's probably not https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/management_group.

Reply:
`terraform/jenkins` is the entrypoint/root terraform module for the current deployment. An additional deployment would need to have its own terraform module (and once we start having more of these, I'd assume we'd factor out more of the common code in reusable modules).

That specific uid won't work in another azure account, but it'd come with its own vault, and where we'd instantiate it we'd decide about access policies there, and which uid should have access. It might be we don't need to give terraform access to the secret at all, if it's not terraform putting it there.

Reply:
ok.